server/single-server/src/main/scala/cool/graph/singleserver/SingleServerDependencies.scala

```diff
@@ -27,44 +26,51 @@ import cool.graph.subscriptions.protocol.SubscriptionRequest
 import cool.graph.system.SystemApiDependencies
+import cool.graph.system.database.Initializers
 import cool.graph.system.database.finder.{CachedProjectResolver, CachedProjectResolverImpl, UncachedProjectResolver}
 import cool.graph.webhook.Webhook
 import cool.graph.websockets.protocol.{Request => WebsocketRequest}
 import cool.graph.worker.payloads.{LogItem, Webhook => WorkerWebhook}
 import cool.graph.worker.services.{WorkerDevServices, WorkerServices}
 import play.api.libs.json.Json
+import slick.jdbc.MySQLProfile
+
+import scala.concurrent.{Await, Future}
 
 trait SingleServerApiDependencies
     extends SystemApiDependencies
     with SimpleApiClientDependencies
     with RelayApiClientDependencies
     with SchemaManagerApiDependencies
     with SimpleSubscriptionApiDependencies {
-
-  override lazy val internalDb              = setupAndGetInternalDatabase()
-  override lazy val kinesis                 = createKinesis()
   override lazy val config                  = ConfigFactory.load()
   override lazy val testableTime            = new TestableTimeImplementation
   override lazy val apiMetricsFlushInterval = 10
-  override lazy val apiMetricsPublisher     = new KinesisPublisherImplementation(streamName = sys.env("KINESIS_STREAM_API_METRICS"), kinesis)
-  override lazy val featureMetricActor      = system.actorOf(Props(new FeatureMetricActor(apiMetricsPublisher, apiMetricsFlushInterval)))
-  override lazy val globalDatabaseManager   = GlobalDatabaseManager.initializeForSingleRegion(config)
   override lazy val clientAuth              = ClientAuthImpl()
-
-  override implicit lazy val bugsnagger = BugSnaggerImpl("")
-
-  override protected def createKinesis(): AmazonKinesis = {
-    val credentials = new BasicAWSCredentials(sys.env("AWS_ACCESS_KEY_ID"), sys.env("AWS_SECRET_ACCESS_KEY"))
-
-    AmazonKinesisClientBuilder
-      .standard()
-      .withCredentials(new AWSStaticCredentialsProvider(credentials))
-      .withEndpointConfiguration(new EndpointConfiguration(sys.env("KINESIS_ENDPOINT"), sys.env("AWS_REGION")))
-      .build()
-  }
+  override implicit lazy val bugsnagger     = BugSnaggerImpl("")
 }
 
 case class SingleServerDependencies(implicit val system: ActorSystem, val materializer: ActorMaterializer) extends SingleServerApiDependencies {
+  import system.dispatcher
+
+  import scala.concurrent.duration._
+
+  val (globalDatabaseManager, internalDb, logsDb) = {
+    val internal = Initializers.setupAndGetInternalDatabase()
+    val logs     = Initializers.setupAndGetLogsDatabase()
+    val client   = Future { GlobalDatabaseManager.initializeForSingleRegion(config) }
+    val dbs      = Future.sequence(Seq(client, internal, logs))
+
+    try {
+      val res = Await.result(dbs, 15.seconds)
+      (res(0).asInstanceOf[GlobalDatabaseManager], res(1).asInstanceOf[MySQLProfile.backend.Database], res(2).asInstanceOf[MySQLProfile.backend.Database])
```

Comment:
To avoid the cast one would probably rather use a for comprehension:
```
for {
  internal <- internalFut
  logs       <- logsFut
  client     <- clientFut
} yield ...
```

Reply:
True, missed that.